src/apiClient.ts

```diff
@@ -161,37 +180,44 @@ export default function APIClient(
         }
     };
 
-    this.sendSingleForwardingStatsToServer = function(forwardingStatsData) {
-        let url;
-        let data;
-        try {
-            const xhrCallback = function() {
-                if (xhr.readyState === 4) {
-                    if (xhr.status === 202) {
-                        mpInstance.Logger.verbose(
-                            'Successfully sent  ' +
-                                xhr.statusText +
-                                ' from server'
-                        );
-                    }
-                }
-            };
-            const xhr = mpInstance._Helpers.createXHR(xhrCallback);
-            url = mpInstance._Helpers.createServiceUrl(
-                mpInstance._Store.SDKConfig.v1SecureServiceUrl,
-                mpInstance._Store.devToken
-            );
-            data = forwardingStatsData;
+    this.sendSingleForwardingStatsToServer = async (
+        forwardingStatsData: IForwardingStatsData
+    ) => {
+        const { v1SecureServiceUrl: forwardingDomain} = mpInstance._Store.SDKConfig;
+        const { devToken } = mpInstance._Store;
+
+        const uploadUrl = `https://${forwardingDomain}${devToken}/Forwarding`;
+
+        let url: string;
+        const uploader: AsyncUploader = window.fetch
+            ? new FetchUploader(uploadUrl)
+            : new XHRUploader(uploadUrl);
+
+        const fetchPayload: fetchPayload = {
+            method: 'post',
+            body: JSON.stringify(forwardingStatsData),
+            headers: {
+                Accept: 'text/plain;charset=UTF-8',
+                'Content-Type': 'text/plain;charset=UTF-8',
+            },
+        };
 
-            if (xhr) {
-                xhr.open('post', url + '/Forwarding');
-                xhr.send(JSON.stringify(data));
-            }
-        } catch (e) {
-            mpInstance.Logger.error(
-                'Error sending forwarding stats to mParticle servers.'
+        const response = await uploader.upload(fetchPayload);
+        
+        // This is a fire and forget, so we only need to log the response based on the code, and not return any response body
+        if (response.status === 202) {
+            mpInstance?.Logger?.verbose(
+                'Successfully sent forwarding stats to mParticle Servers configuration from server'
+            );
+        } else {
+            mpInstance?.Logger?.verbose(
+                'Issue with receiving configuration from server, received HTTP Code of ' +
+                    response.statusText
             );
         }
+
```

Comment:
I think this makes things more readable and possibly can remove the need for a comment.

```suggestion
        // This is a fire and forget, so we only need to log the response based on the code, and not return any response body
        let message: string;
        if (response.status === 202) {
            message = 'Successfully sent forwarding stats to mParticle Servers configuration from server';
        } else {
            message = 'Issue with receiving configuration from server, received HTTP Code of ' + response.statusText;
        }
        mpInstance?.Logger?.verbose(message);
```